lib/tree.rb

```diff
@@ -1,25 +1,41 @@
 class NoApplesError < StandardError; end
 
-class AppleTree
-  attr_#fill_in :height, :age, :apples, :alive
+class Tree
+  attr_reader :height, :age
+  MAXIMUM_AGE = 100
+  MAXIMUM_FRUIT_DIAMETER = 10
 
   def initialize
+    @height = 0.5
+    @age = 0
+    @apples = []
   end
 
   def age!
+    @age += 1
+    @height += 0.5 unless height >= 30
+    add_apples
   end
 
   def add_apples
+    if @age >= 3
+      rand(@age / 2...@age).times do
+        @apples.push(Apple.new('red', rand(MAXIMUM_FRUIT_DIAMETER)))
+      end
+    end
   end
 
   def any_apples?
+    @apples.length > 0
   end
 
   def pick_an_apple!
     raise NoApplesError, "This tree has no apples" unless self.any_apples?
+    @apples.pop
   end
 
   def dead?
+    @age >= MAXIMUM_AGE
```

Comment:
Its good practice to refer to the getter method in as many cases as possible. Imagine a scenario in which you have to make modifications to an instance variable before utilizing it in other methods.